### ISSUE TITLE
feat: Improve documentation about adding the Firebase SDK

### ISSUE THREAD

**Issue by basvdijk** [capawesome-team/capacitor-firebase#205]
**Is your feature request related to a problem? Please describe:**

Most Capacitor plugins do not require to add an SDK to the iOS or Android project itself.

**Describe the solution you'd like:**

A clear instruction how to add the SDK to my project. The firebase documentation to which is linked assumes you have a 100% native app. For example on iOS it is unclear for me to which project I have to add the Firebase SDK. The app project or the pod project?

**Describe alternatives you've considered:**

**Additional context:**

**Comment by basvdijk** [comment]
@robingenz I decided to write my own documentation for Android and iOS. Also added a demo project in order to test Firebase Analytics screen changes. Maybe my documentation could be helpful to other developers: https://github.com/basvdijk/capacitor4-firebase-analytics-example-project-capawesome-team

**Comment by robingenz** [comment]
@basvdijk Thank you! I will use this as inspiration.

**Comment by ubaranzorlu** [comment]
Support

**Comment by piotr-cz** [comment]
Too bad that I have found this issue just *after* hours of debugging of what's wrong.

The thing is to skip the * Add Firebase SDKs to your app* and *Initialize Firebase in your app* steps from [iOS](https://firebase.google.com/docs/ios/setup) guide

**Comment by robingenz** [comment]
> The thing is to skip the * Add Firebase SDKs to your app* and Initialize Firebase in your app steps from the iOS guide

Sorry for that. I have just seen that Firebase has reworked the documentation. I will update the installation instructions.

**Comment by robingenz** [comment]
@piotr-cz I updated the docs: https://github.com/capawesome-team/capacitor-firebase/pull/316

**Comment by piotr-cz** [comment]
@robingenz Thank you for preparing updated instructions, I'm sure it would help new devs!

**Comment by szymon-mutuu** [comment]
isn't this plugin supposed to configure all the sdk? how to proceed for cicd?

**Comment by robingenz** [comment]
> isn't this plugin supposed to configure all the sdk?

@szymon-mutuu No, this plugin allows you to use the Firebase SDKs in Capacitor. But you have to configure the SDKs yourself.

> how to proceed for cicd?

Nothing special. You configure the Android and iOS projects during setup according to the documentation and push the state to git. The CI then has all the information it needs. If for some reason you don't want to push files such as `GoogleService-Info.plist` to git, then you can easily add them via CI before the build ([example](https://github.com/robingenz/capacitor-firebase-plugin-demo/blob/e1684a0af6871442ed0a87dceeeba6fd9ce0185d/.github/workflows/ci.yml#L87)). Nothing special here either.

If you have more questions, please create a discussion. These questions are off topic.

**Comment by szymon-mutuu** [comment]
thanks @robingenz for prompt response! 
Am I correct it assumes keeping ios project in the repo? 🤔 I don't have ios/android projects in the repo and all works well so far for pretty complex project. I think pushing ios project to the repo is not a good idea since there is little control over any changes to binary files being consequence of using Xcode...

I think there should be a way to make this automated and that there were some eg cordova projects installing firebase sdk in a headless way. I appreciate existence of this project, just loudly thinking how it possibly be improved...

**Comment by robingenz** [comment]
> If you have more questions, please create a discussion. These questions are off topic.

@szymon-mutuu As already said, please create a discussion and link your comment.

**Comment by szymon-mutuu** [comment]
sure, also here for reference https://github.com/capawesome-team/capacitor-firebase/discussions/338
